fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java

```diff
@@ -570,17 +570,19 @@ public static boolean isFailure(IssueSeverity severity) {
     }
 
     /**
-     * Generate a random AES key or 32 byte value encoded as a Base64 string.
+     * Generate a random key using the passed algorithm or, if that algorithm isn't supported, a random 32 byte value.
+     * In either case, the resulting value is encoded as a Base64 string before returning.
      *
-     * @return
+     * @return a base64-encoded random key string
      */
-    public static String getRandomKey(String key) {
+    public static String getRandomKey(String algorithm) {
         KeyGenerator keyGen;
         try {
-            keyGen = KeyGenerator.getInstance(key);
+            keyGen = KeyGenerator.getInstance(algorithm);
             keyGen.init(256);
             return Base64.getEncoder().encodeToString(keyGen.generateKey().getEncoded());
         } catch (NoSuchAlgorithmException e) {
+            log.warning("Algorithm '" + algorithm + "' is not supported; using SecureRandom instead");
             byte[] buffer = new byte[32];
             RANDOM.setSeed(System.currentTimeMillis());
```

Comment:
I know it's not related to this change, but why are we seeding here? Potential OWASP because currentTimeMillis isn't sufficiently random.

Reply:
I was wondering the same thing, which is why i added the warning...i'm not even sure it should fallback to SecureRandom

Reply:
I don't think this should be in FHIRUtil -- it should go into bulkdata, and be hidden away.